### PR TITLE
Avoid `Testem.hookIntoTestFramework` unless `QUnit` or `Mocha` are defined.

### PR DESCRIPTION
**Pull request by rwjblue**
Fixes https://github.com/embroider-build/embroider/issues/530

### CODE REVIEW

File: packages/core/src/app.ts [hunk 1]
```diff
@@ -253,8 +253,8 @@ export class AppBuilder<TreeNames> {
         relativePath: '_testing_suffix_.js',
         source: `
         var runningTests=true;
-        if (window.Testem) {
-          window.Testem.hookIntoTestFramework();
+        if (typeof Testem !== 'undefined' && (typeof QUnit !== 'undefined' || typeof Mocha !== 'undefined')) {
+          Testem.hookIntoTestFramework();
         }`,
```
Comment (rwjblue) [inline packages/core/src/app.ts:258]: FWIW, I think this is transitionary. Once I can roll out ember-qunit@5.0.0 more broadly, we can remove this completely (and release a breaking change release).

**Comment by rwjblue** [comment]
I've confirmed (by locally editing `node_modules/@embroider/core/src/app.js`) that https://github.com/emberjs/ember-qunit/pull/756 passes with these changes.

**Comment by ef4** [comment]
Thanks.

If QUnit or Mocha is not defined at this point, where does the `hookIntoTestFramework()` thing happen instead?

**Comment by rwjblue** [comment]
> If QUnit or Mocha is not defined at this point, where does the hookIntoTestFramework() thing happen instead?

`ember-qunit` invokes it when it knows for sure that QUnit is defined (because it itself has imported it):

https://github.com/emberjs/ember-qunit/blob/32c4cee9dd266bf36f76274d33c713306feada61/addon-test-support/index.js#L8-L10

(Added in https://github.com/emberjs/ember-qunit/pull/752/commits/223e0cf3cad9ec3e700125e81da40c3dbd588c67)

**Pull request merged**